### PR TITLE
refact(webhook): add validation for new fields in cspc

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

This PR
-  adds validation for new schema changes in CSPC. 
-  reverts the API version of `ValidatingWebhookConfig` to v1beta1 as the `sideEffectClass` `NoneOnDryRun` does not seem to work in this case and v1 does not support `Unknown` class. Will look into it until then using v1beta1.

TODO:
 - validation for expansion
 - validation for capacity

will take these in separate PRs.

### CODE REVIEW

File: pkg/webhook/cspc.go [hunk 1]
```diff
@@ -262,11 +318,17 @@ var (
...
+		cstor.PoolStriped:  isStripedBDCountValid,
+		cstor.PoolMirrored: isMirroredBDCountValid,
+		cstor.PoolRaidz:    isRaidzBDCountValid,
+		cstor.PoolRaidz2:   isRaidz2BDCountValid,
+	}
+	// SupportedCompression is a map holding the supported compressions
+	SupportedCompression = map[string]bool{
+		"":    true,
+		"off": true,
+		"lz":  true,
```
Comment (mittachaitu) [inline pkg/webhook/cspc.go]: Not sure what and all are supporting
```c
 static zprop_index_t compress_table[] = {
                { "on",         ZIO_COMPRESS_ON },
                { "off",        ZIO_COMPRESS_OFF },
                { "lzjb",       ZIO_COMPRESS_LZJB },
                { "gzip",       ZIO_COMPRESS_GZIP_6 },  /* gzip default */
                { "gzip-1",     ZIO_COMPRESS_GZIP_1 },
                { "gzip-2",     ZIO_COMPRESS_GZIP_2 },
                { "gzip-3",     ZIO_COMPRESS_GZIP_3 },
                { "gzip-4",     ZIO_COMPRESS_GZIP_4 },
                { "gzip-5",     ZIO_COMPRESS_GZIP_5 },
                { "gzip-6",     ZIO_COMPRESS_GZIP_6 },
                { "gzip-7",     ZIO_COMPRESS_GZIP_7 },
                { "gzip-8",     ZIO_COMPRESS_GZIP_8 },
                { "gzip-9",     ZIO_COMPRESS_GZIP_9 },
                { "zle",        ZIO_COMPRESS_ZLE },
                { "lz4",        ZIO_COMPRESS_LZ4 },
                { NULL }
        };
```
cc: @sonasingh46

Comment (mittachaitu) [reply pkg/webhook/cspc.go]: @shubham14bajpai We need to support above all compressions.

Comment (mittachaitu) [reply pkg/webhook/cspc.go]: As of know, the existing map is good. We will revisit if we need to support more...

Comment (shubham14bajpai) [reply pkg/webhook/cspc.go]: Added a TODO comment for this. Once we are sure we can update the map.

File: pkg/webhook/cspc.go [hunk 2]
```diff
@@ -285,17 +362,11 @@ func (poolValidator *PoolValidator) raidGroupValidation(
 	raidGroup *cstor.RaidGroup, rgType string) (bool, string) {
 
 	if len(raidGroup.BlockDevices) == 0 {
-		return false, fmt.Sprintf("number of block devices honouring raid type should be specified")
```
Comment (mittachaitu) [inline pkg/webhook/cspc.go]: Below check is good enough we don't require this check(We can remove in next PR).

**Pull request merged**